libs/javascript/metadatafetcher.js

```diff
@@ -0,0 +1,107 @@
+'use strict';
+
+const { createTokenAuth } = require("@octokit/auth-token");
+const fetch = require('node-fetch');
+const fs = require('node:fs');
+const { mkdir } = require('node:fs/promises');
+const path = require('node:path');
+const { request } = require("@octokit/request");
+
+async function createDirIfNotExisting(dirPath) {
+
+    if (!fs.existsSync(dirPath)) {
+
+        await mkdir(dirPath);
+
+    }
+}
+
+async function getContent() {
+
+    const sourcePath = `${process.env["METADATA_SOURCE_PATH"]}/${process.env["METADATA_SOURCE_VERSION"]}`;
+    const targetPath = process.env["METADATA_TARGET_PATH"];
+
+    const result = await getContentFromGitHub(sourcePath);
+
+    await storeResultInRepo(result, targetPath);
+}
+
+async function getContentFromGitHub(sourcePath) {
+
+    const pat = process.env["METADATA_READ_TOKEN"];
+    const owner = "btp-setup-automator";
+    const repo = "btp-metadata-fetcher";
```

Comment:
Should this information be public? If not, it should be provided via environmental.

Reply:
Makes sense to parameterize these parameters